fw/http.c

```diff
@@ -5627,6 +5627,18 @@ __extract_request_authority(TfwHttpReq *req)
 static bool
 __check_authority_correctness(TfwHttpReq *req)
 {
+	int conn_type = TFW_CONN_TYPE(req->conn)
+			& (TFW_FSM_HTTPS | TFW_FSM_WEBSOCKET);
+	if (conn_type == TFW_FSM_HTTPS) {
+		TlsCtx *tctx = tfw_tls_context(req->conn);
+
+		if (likely(tctx->sni != NULL) &&
+		    unlikely(!tfw_str_eq_cstr(&req->host,
+		                              tctx->sni, tctx->sni_len, 0))) {
```

Comment:
Probably just to add a pointer to vhost to `TlsCtx` (it could be `void *` to not to mess up the TLS layer with HTTP) and compare `vhost->name` with `req->host` here? Also we can replace this check just with comparison of vhost pointers after HTTP tables call in `tfw_http_req_process()`. However, the host comparison shouldn't be heavy though....